### ISSUE TITLE
Implementation of close to tray, while minimize not to tray

### ISSUE THREAD

**Issue by knakamura8** [Betterbird/thunderbird-patches#244]
I think that there should be a feature by which BB can run persistently in the system tray (at least in Windows), while retaining normal minimize to taskbar (not tray) functionality.

I am aware of the built-in functionality of minimize to tray, built-in persistent tray icon, as well as the close to minimize extension.  I have also read tickets #190 and #111.

The typical workflow for apps that support close to tray, imho, is that minimize functionality works normally (i.e. to taskbar), and close will close the app to the system tray.  Currently, the user has to choose between whether or not to have taskbar functionality with no close to tray, or close to tray with no taskbar functionality (via the chaining of close to minimize -> minimize to tray -> tray icon, even with always persist tray icon enabled).

My proposal is an inclusion of close to tray, which functions independently of minimize to tray.  This way, users can choose which functionality (maybe even both!), they wish to enable.  As such, I propose the addition of a column on the System tray matrix, of the nature of Close to tray.  i.e.

OS / Desktop | System tray | System tray always | Tooltip with folder list | Minimize to tray | Close to tray
-- | -- | -- | -- | -- | --
Windows | Yes | Yes | Yes | Yes | Hopefully!
Linux / KDE | Yes | Yes | Yes | Yes | Hopefully!
Linux / Gnome | Yes | Yes | No | Yes(*) | Hopefully!
Linux / Cinnamon | Yes | Yes | Yes | No | Hopefully!
Linux / Mate | Yes | Yes | Yes | Yes(**) | Hopefully!
Linux / Xfce | Yes | Yes | Yes | Yes | Hopefully!

**Comment by Betterbird** [comment]
So basically you want to have what this add-on https://addons.thunderbird.net/en-us/thunderbird/addon/minimize-on-close/ does "out of the box", but the "close minimise" should go to the tray always. Right?

**Comment by knakamura8** [comment]
Well, considering that the extension already handles minimize on close, of course it would not be strictly necessary to re-implement that within BB directly, though it would be "nice to have", imho.

The workflow suggestion is more that I think close to tray is useful for apps that run persistently, such as an email program which notifies the user of incoming mail.  Chat, RSS, and file transfer programs are other type programs which tend to support this type of Close to tray feature, while retaining normal minimize to taskbar functionality.  For example, if a user is tabbing between taskbar windows, including minimizing them when writing an email or using a program such as a chat program, then having the window be minimized to tray is inconvenient.  However, upon completion of the active tasks at hand, it is convenient to close the program to tray, and still have the program waiting on incoming emails, messages, feeds, files, etc.

Currently, when using that extension, it essentially must be chained to minimize to tray to allow for "close to tray" to work.  Whereas the workflow I describe divorces the two functions, making them independent of one another.  I apologize if I am not expressing myself clearly.  Also, I am unsure as to how much complexity this might entail.  That having been said, it is a common feature in such "persistent" type programs, both in Windows and in GNU/Linux distributions.  A stretch/reach goal would perhaps to just include both minimize to tray and close to tray functionality as separate check boxes within BB itself, which would also obviate needing the extension at all.

**Comment by Betterbird** [comment]
Why don't you suggest to the author of the "Minimize on close" @rsjtdrjgfuzkfg that he implement an option to "minimize to tray on close"? That would be much less coding effort that us having to integrate his add-on and then add this option?

**Comment by knakamura8** [comment]
OK, I will do that, thank you!

**Comment by Betterbird** [comment]
We haven't had any further feedback here. Looks like using the add-on plus the "minimise to tray" function in BB already does what you want. We tested it on Windows.

Note that "minimise to tray" doesn't work on Wayland, see Issue #279.

**Comment by slikie** [comment]
Wayland/Sway/Hyprland doesn't seem to work with either of these?

**Comment by Betterbird** [comment]
Anything Wayland doesn't work, see Issue #279.

**Comment by Lachcim** [comment]
Good news! I was able to implement this feature as an extension. It's called **Close to Tray**, you can get the Windows-only version from [addons.thunderbird.net here](https://addons.thunderbird.net/en-US/thunderbird/addon/close-to-tray/) and the Betterbird-enabled version from [the GitHub page here](https://github.com/Lachcim/thunderbird-close-to-tray/).

My original plan was to have a single version that works with Betterbird on Linux, but the Mozilla folks claimed that openly bringing up Betterbird went against their addon review policy, specifically the part that prohibits promoting third-party products. I was asked to remove any mentions of Betterbird from my addon and its listing. That's a shame.

**Comment by Betterbird** [comment]
Nice!

**Comment by Betterbird** [comment]
Thunderbird decided to integrate the "Close to Tray" add-on in their https://bugzilla.mozilla.org/show_bug.cgi?id=2044250 in TB 154.

Unfortunately they removed "Minimise to tray" at the same time. We don't agree with this decision. Additionally, we had already implemented "start minimised" here: https://github.com/Betterbird/thunderbird-patches/blob/main/140/features/19-feature-startup-minimized.patch behind pref `mail.startupMinimized`.

So in Betterbird 153 we're bringing forward Thunderbird implementation of "Close to Tray", but we restored "Minimise to Tray" and we're using the existing pref for "start minimised" which they sadly connected to "Close to Tray" due to a management override of another manager's opinion and the [opinion of a senior developer](https://phabricator.services.mozilla.com/D304012#10568999).

In total, it looks like this in BB 153, all three options are independent:

<img width="437" height="204" alt="Image" src="https://github.com/user-attachments/assets/c8e2d737-0f03-46ee-8f09-40c018acfbea" />

https://github.com/Betterbird/thunderbird-patches/commit/c35eb1d8e2c213cb58eaecaf19dd808e721a4941 https://github.com/Betterbird/thunderbird-patches/commit/275b3397588582a829a4377d0585c16f9cb812af https://github.com/Betterbird/thunderbird-patches/commit/39c8f787a13821028cb4d13e2b447a8537f8849b

And of course it works for Linux, too. The Thunderbird folks are making their lives hard since they still have "minimise to tray" for Linux [on file](https://bugzilla.mozilla.org/show_bug.cgi?id=1627479), yet, the function has just been dumped on Windows. Well, one hand doesn't know what the other hand does.
